js/lib/edge.js

```diff
@@ -728,11 +727,11 @@ function getDefaultService() {
 }
 
 function createServiceFromCapabilities(options) {
-  let exe;
-  if (useEdgeChromium(options)){
-    exe = locateSynchronously(EDGE_CHROMIUM_BROWSER_NAME);
+  if (useEdgeChromium(options)) {
+    let exe = locateSynchronously(EDGE_CHROMIUM_BROWSER_NAME);
+    return new ServiceBuilder(exe).build();
   }
-  return new ServiceBuilder(exe).build();
+  return getDefaultService();
```

Comment:
The default service could point to either a legacy edge service or a chromium edge service. When the default service points to a chromium edge service and user would like to create a legacy edge drive, this createSession request will fail in this situation.

Reply:
This is to preserve the existing behavior as much as possible: if the user sets a default service and creates a driver without passing a service, then the default service will be used.

If the user is providing their own service (either through setDefaultService or opt_service), they are responsible for providing the correct kind.